src/transformers/models/phi/modular_phi.py

```diff
@@ -0,0 +1,1066 @@
+        hidden_size=2048,
+        intermediate_size=8192,
+        num_hidden_layers=24,
+        num_attention_heads=32,
+        num_key_value_heads=None,
+        resid_pdrop=0.0,
+        embd_pdrop=0.0,
+        attention_dropout=0.0,
+        hidden_act="gelu_new",
+        max_position_embeddings=2048,
+        initializer_range=0.02,
+        layer_norm_eps=1e-5,
+        use_cache=True,
+        tie_word_embeddings=False,
+        rope_theta=10000.0,
+        rope_scaling=None,
+        partial_rotary_factor=0.5,
+        qk_layernorm=False,
+        bos_token_id=1,
+        eos_token_id=2,
+        **kwargs,
+    ):
+        self.vocab_size = vocab_size
+        self.hidden_size = hidden_size
+        self.intermediate_size = intermediate_size
+        self.num_hidden_layers = num_hidden_layers
+        self.num_attention_heads = num_attention_heads
+
+        if num_key_value_heads is None:
+            num_key_value_heads = num_attention_heads
+
+        self.num_key_value_heads = num_key_value_heads
+        self.resid_pdrop = resid_pdrop
+        self.embd_pdrop = embd_pdrop
+        self.attention_dropout = attention_dropout
+        self.hidden_act = hidden_act
+        self.max_position_embeddings = max_position_embeddings
+        self.initializer_range = initializer_range
+        self.layer_norm_eps = layer_norm_eps
+        self.use_cache = use_cache
+        self.rope_theta = rope_theta
+        self.rope_scaling = rope_scaling
+        self.partial_rotary_factor = partial_rotary_factor
+        self.qk_layernorm = qk_layernorm
+        # Validate the correctness of rotary position embeddings parameters
+        # BC: if there is a 'type' field, move it to 'rope_type'.
+        if self.rope_scaling is not None and "type" in self.rope_scaling:
+            self.rope_scaling["rope_type"] = self.rope_scaling["type"]
+        rope_config_validation(self)
+
+        super().__init__(
+            bos_token_id=bos_token_id,
+            eos_token_id=eos_token_id,
+            tie_word_embeddings=tie_word_embeddings,
+            **kwargs,
+        )
+
+
+class PhiRotaryEmbedding(LlamaRotaryEmbedding):
+    pass
+
+
+class PhiLinearScalingRotaryEmbedding(LlamaLinearScalingRotaryEmbedding, PhiRotaryEmbedding):
+    pass
+
+
+class PhiDynamicNTKScalingRotaryEmbedding(LlamaDynamicNTKScalingRotaryEmbedding, PhiRotaryEmbedding):
+    pass
+
+
+class PhiMLP(nn.Module):
+    def __init__(self, config):
+        super().__init__()
+        self.config = config
+        self.activation_fn = ACT2FN[config.hidden_act]
+        self.fc1 = nn.Linear(config.hidden_size, config.intermediate_size)
+        self.fc2 = nn.Linear(config.intermediate_size, config.hidden_size)
+
+    def forward(self, hidden_states: torch.Tensor) -> torch.Tensor:
+        hidden_states = self.fc1(hidden_states)
+        hidden_states = self.activation_fn(hidden_states)
+        hidden_states = self.fc2(hidden_states)
+        return hidden_states
```

Comment:
I had initially tried it. But This comes at tradeof of RENAMING to `PHIMLP` than `PhiMLP`. Is that Okay? Otherwise the `modular_model_converted.py > ModularConverterTransformer > leave_ClassDef` needs to be fixed to handle such cases.

Reply:
I see. This will be fixed by https://github.com/huggingface/transformers/pull/34487! The PR should be merged very soon if you can wait just a little bit. After it is merged, if you still experience this issue let me know!

Reply:
Yeah. Thanks for updating. 
One more thing. I've tried to convert other models to modular, and am suffering with double import of `_flash_attention_forward` One in if block and other in global scope. Including the current `modular_glm.py`
I hope the new modular_model_converter solves it as well.
I'll probably fork or pull from your PR and test it.

Reply:
Thenks For the PR, Now it stays `PhiMLP`